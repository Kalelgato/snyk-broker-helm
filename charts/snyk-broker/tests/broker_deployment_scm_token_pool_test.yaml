suite: test broker deployment (credential pooling)
chart:
  version: 0.0.0
templates:
  - broker_deployment.yaml
  - broker_service.yaml
  - secrets.yaml
  - serviceaccount.yaml
values:
  - ./fixtures/default_values.yaml

tests:
  - it: github token pool configured
    values:
      - ./fixtures/github_tokenpool.yaml
    asserts:
      - matchSnapshot: {}
  - it: gitlab token pool configured
    values:
      - ./fixtures/gitlab_tokenpool.yaml
    asserts:
      - matchSnapshot: {}
  - it: github token pool configured with enabled useExternalSecretScmTokenPool
    values:
      - ./fixtures/github_tokenpool_external_secret.yaml
    asserts:
      - matchSnapshot: {}
