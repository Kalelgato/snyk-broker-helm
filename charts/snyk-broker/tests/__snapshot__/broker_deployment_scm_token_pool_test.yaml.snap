github token pool configured:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: github-com-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        spec:
          containers:
            - env:
                - name: BROKER_SERVER_URL
                  value: https://broker.test.snyk.io
                - name: BROKER_HEALTHCHECK_PATH
                  value: /healthcheck
                - name: BROKER_SYSTEMCHECK_PATH
                  value: /systemcheck
                - name: BROKER_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: github-com-broker-token-key
                      name: github-com-broker-token-RELEASE-NAME-snyk-broker
                - name: GITHUB_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: github-com-token-key
                      name: github-com-token-RELEASE-NAME-snyk-broker
                - name: GITHUB_TOKEN_POOL
                  valueFrom:
                    secretKeyRef:
                      key: github-com-token-key-pool
                      name: github-com-token-pool-RELEASE-NAME-snyk-broker
                - name: PORT
                  value: "8000"
                - name: BROKER_CLIENT_URL
                  value: http://brokerclient
                - name: LOG_LEVEL
                  value: info
                - name: LOG_ENABLE_BODY
                  value: "false"
                - name: ACCEPT_CODE
                  value: "true"
                - name: ACCEPT_IAC
                  value: tf,yaml,yml,json,tpl
                - name: BROKER_DISPATCHER_BASE_URL
                  value: https://api.test.snyk.io
              image: snyk/broker:github-com
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              name: github-com-broker
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 1
                  memory: 256Mi
                requests:
                  cpu: 1
                  memory: 256Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts: null
          securityContext: {}
          serviceAccountName: snyk-broker
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: github-com-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      type: ClusterIP
  3: |
    apiVersion: v1
    data:
      github-com-broker-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: github-com-broker-token-RELEASE-NAME-snyk-broker
    type: Opaque
  4: |
    apiVersion: v1
    data:
      github-com-token-key-pool: Z2hfdG9rZW4xLGdoX3Rva2VuMixnaF90b2tlbjM=
    kind: Secret
    metadata:
      name: github-com-token-pool-RELEASE-NAME-snyk-broker
    type: Opaque
  5: |
    apiVersion: v1
    data:
      snyk-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: snyk-token-RELEASE-NAME-snyk-broker
    type: Opaque
  6: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: snyk-broker
      namespace: NAMESPACE
github token pool configured with enabled useExternalSecretScmTokenPool:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: github-com-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        spec:
          containers:
            - env:
                - name: BROKER_SERVER_URL
                  value: https://broker.test.snyk.io
                - name: BROKER_HEALTHCHECK_PATH
                  value: /healthcheck
                - name: BROKER_SYSTEMCHECK_PATH
                  value: /systemcheck
                - name: BROKER_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: github-com-broker-token-key
                      name: github-com-broker-token-RELEASE-NAME-snyk-broker
                - name: GITHUB_TOKEN_POOL
                  valueFrom:
                    secretKeyRef:
                      key: github-com-token-key-pool
                      name: github-com-token-pool-RELEASE-NAME-snyk-broker
                - name: PORT
                  value: "8000"
                - name: BROKER_CLIENT_URL
                  value: http://brokerclient
                - name: LOG_LEVEL
                  value: info
                - name: LOG_ENABLE_BODY
                  value: "false"
                - name: ACCEPT_CODE
                  value: "true"
                - name: ACCEPT_IAC
                  value: tf,yaml,yml,json,tpl
                - name: BROKER_DISPATCHER_BASE_URL
                  value: https://api.test.snyk.io
              image: snyk/broker:github-com
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              name: github-com-broker
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 1
                  memory: 256Mi
                requests:
                  cpu: 1
                  memory: 256Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts: null
          securityContext: {}
          serviceAccountName: snyk-broker
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: github-com-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      type: ClusterIP
  3: |
    apiVersion: v1
    data:
      github-com-broker-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: github-com-broker-token-RELEASE-NAME-snyk-broker
    type: Opaque
  4: |
    apiVersion: v1
    data:
      snyk-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: snyk-token-RELEASE-NAME-snyk-broker
    type: Opaque
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: snyk-broker
      namespace: NAMESPACE
gitlab token pool configured:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: gitlab-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        spec:
          containers:
            - env:
                - name: BROKER_SERVER_URL
                  value: https://broker.test.snyk.io
                - name: BROKER_HEALTHCHECK_PATH
                  value: /healthcheck
                - name: BROKER_SYSTEMCHECK_PATH
                  value: /systemcheck
                - name: BROKER_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: gitlab-broker-token-key
                      name: gitlab-broker-token-RELEASE-NAME-snyk-broker
                - name: GITLAB_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: gitlab-token-key
                      name: gitlab-token-RELEASE-NAME-snyk-broker
                - name: GITLAB_TOKEN_POOL
                  valueFrom:
                    secretKeyRef:
                      key: gitlab-token-key-pool
                      name: gitlab-token-pool-RELEASE-NAME-snyk-broker
                - name: GITLAB
                  value: null
                - name: PORT
                  value: "8000"
                - name: BROKER_CLIENT_URL
                  value: http://brokerclient
                - name: LOG_LEVEL
                  value: info
                - name: LOG_ENABLE_BODY
                  value: "false"
                - name: ACCEPT_CODE
                  value: "true"
                - name: ACCEPT_IAC
                  value: tf,yaml,yml,json,tpl
                - name: BROKER_DISPATCHER_BASE_URL
                  value: https://api.test.snyk.io
              image: snyk/broker:gitlab
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              name: gitlab-broker
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 1
                  memory: 256Mi
                requests:
                  cpu: 1
                  memory: 256Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts: null
          securityContext: {}
          serviceAccountName: snyk-broker
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: gitlab-RELEASE-NAME-snyk-broker
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
      type: ClusterIP
  3: |
    apiVersion: v1
    data:
      gitlab-broker-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: gitlab-broker-token-RELEASE-NAME-snyk-broker
    type: Opaque
  4: |
    apiVersion: v1
    data:
      gitlab-token-key-pool: Z2xfdG9rZW5fMSxnbF90b2tlbl8y
    kind: Secret
    metadata:
      name: gitlab-token-pool-RELEASE-NAME-snyk-broker
    type: Opaque
  5: |
    apiVersion: v1
    data:
      snyk-token-key: MTIzNDVhNmItNzhkOS0xZTJmLTNhYmMtNDU2ZGVmNzg5MTIz
    kind: Secret
    metadata:
      name: snyk-token-RELEASE-NAME-snyk-broker
    type: Opaque
  6: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-snyk-broker
        helm.sh/chart: snyk-broker-0.0.0
      name: snyk-broker
      namespace: NAMESPACE
